semanticdb/semanticdb3/semanticdb3.md

```diff
@@ -2325,6 +2447,249 @@ language constructs and idioms. However, but we do not yet have a specification
 that comprehensively describes how Scala language features map onto symbol
 occurrences. We intend to improve on this in the future.
 
+<a name="scala-synthetic"></a>
+
+#### Synthetic
+
+Here is an exhaustive list of the kinds of synthetics that are generated by the
+scalac plugin, along with examples:
+
+<table>
```

Comment:
A) Please provide raw ADTs for all synthetics except for loops, like it's done for types. Otherwise, we'd have to provide a specification for Metap, which is much more work. E.g. the first example would become:

```
Synthetic(
  <List>, 
  TypeApplyTree(
    SelectTree(
      OriginalTree(<List>),
      IdTree(<List.apply>)),
    List(
      IdTree(<Int>))))
```

Reply:
B) I'd suggest to drop `flatMap => scala/collection/TraversableLike#flatMap().` and the like. I don't think they help understand synthetics, which is the goal of this section of the spec, so we might as well make things more concise by not showing them.

Reply:
C) In order to explain what's going on with for loops, preface the table with something along the lines of: "In the examples below, synthetics generated to represent for loops are quite massive, so we are using a shorthand notation for them. In that notation, OriginalTree trees are represented by `orig(<code>)` and other trees are represented by their syntax.`

Reply:
D) Also, I think that `* => ` in for loop examples should be dropped.